delta-lake/common/src/main/databricks/scala/com/nvidia/spark/rapids/delta/DatabricksDeltaProvider.scala

```diff
@@ -116,6 +117,15 @@ object DatabricksDeltaProvider extends DeltaProviderImplBase {
     catalogClass == classOf[DeltaCatalog] || catalogClass == classOf[UnityCatalogV2Proxy]
   }
 
+  private def getWriteOptions(options: Any): Map[String, String] = {
+    // For Databricks 13.3 AtomicCreateTableAsSelectExec writeOptions is a Map[String, String]
+    // while in all the other versions it's a CaseInsensitiveMap
+    options match {
+      case c: CaseInsensitiveStringMap => c.asCaseSensitiveMap().asScala.toMap
+      case _ => options.asInstanceOf[Map[String, String]]
```

Comment:
This won't work due to type erasure. Given that this is a private method why do you think we should protect against getting an unexpected type?

Reply:
oh right forgot about the type erasure on matching.  It's fine as-is.  The check added at the end was to make the error message a bit more useful in case it didn't match.  Not expected, but stranger things have happened.

>  Given that this is a private method why do you think we should protect against getting an unexpected type?

It was not so much to protect (the cast in the original code could also throw) but to make it a bit nicer of an error message if it did happen.  We're pulling from a CPU exec that could theoretically change the type from what we compiled with, so it's not really about the method being private or not but instead where the argument is coming from.